scipy/stats/_multivariate.py

```diff
@@ -3825,3 +3828,401 @@ def rvs(self, dim, size=1, random_state=None):
 
 
 unitary_group = unitary_group_gen()
+
+
+_mvt_doc_default_callparams = \
+"""
+mean : array_like, optional
+    Mean of the distribution. (default ``0``)
+shape : array_like, optional
+    Positive definite matrix of the distribution. (default ``1``)
+df : integer, optional
+    Degrees of freedom of the distribution; must be a positive whole number. (default ``1``) 
+allow_singular : bool, optional
+    Whether to allow a singular matrix. (default ``False``)
+"""
+
+_mvt_doc_callparams_note = \
+"""Setting the parameter `mean` to ``None`` is equivalent to having `mean`
+be the zero-vector. The parameter `shape` can be a scalar, in which case
+the shape matrix is the identity times that value, a vector of
+diagonal entries for the shape matrix, or a two-dimensional array_like.
+"""
+
+_mvt_doc_frozen_callparams_note = \
+"""See class definition for a detailed description of parameters."""
+
+mvt_docdict_params = {
+    '_mvt_doc_default_callparams': _mvt_doc_default_callparams,
+    '_mvt_doc_callparams_note': _mvt_doc_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+mvt_docdict_noparams = {
+    '_mvt_doc_default_callparams': "",
+    '_mvt_doc_callparams_note': _mvt_doc_frozen_callparams_note,
+    '_doc_random_state': _doc_random_state
+}
+
+
+class multivariate_t_gen(multi_rv_generic):
+    r"""
+    A multivariate t-distributed random variable.
+
+    The `mean` parameter specifies the mean. The `shape` parameter specifies 
+    the positive definite shape matrix. The `df` parameter specifies the 
+    degrees of freedom.
+
+    In addition to calling the methods below, the object itself may be called 
+    as a function to fix the mean, shape matrix, and degrees of freedom 
+    parameters, returning a "frozen" multivariate t-distribution random.
+
+    .. versionadded:: 1.5.0
+
+    Methods
+    -------
+    ``pdf(x, mean=None, shape=1, df=1, allow_singular=False)``
+        Probability density function.
+    ``logpdf(x, mean=None, shape=1, df=1, allow_singular=False)``
+        Log of the probability density function.
+    ``rvs(mean=None, shape=1, df=1, size=1, random_state=None)``
+        Draw random samples from a multivariate t-distribution.
+
+    Parameters
+    ----------
+    x : array_like
+        Quantiles, with the last axis of `x` denoting the components.
+    %(_mvt_doc_default_callparams)s
```

Comment:
Interesting, I don't remember the last time I saw this substitution approach to docstrings, but maybe this is common these days.

Reply:
I'm happy to modify this if there's a more standard way of doing it. I just copied the existing code in `_multivariate.py`. CTRL+F for `_mvn_doc_default_callparams` to find that example.

Reply:
It's fine as is. It's not that common in the whole SciPy code base, but for distributions in `stats` it's quite handy and done a lot. Also in `ndimage`.